packages/opentelemetry-instrumentation-http/test/functionals/http-enable.test.ts

```diff
@@ -480,12 +480,26 @@ describe('HttpInstrumentation', () => {
       }
 
       it('should have 1 ended span when request throw on bad "options" object', () => {
-        try {
-          http.request({ protocol: 'telnet' });
-        } catch (error) {
+        assert.throws(() => http.request({ headers: { cookie: undefined} }), err => {
           const spans = memoryExporter.getFinishedSpans();
```

Comment:
I think this code part must be moved out of the callback as I think it's never called. so it should be like this:
```js
assert.throws(() => http.request({ headers: { cookie: undefined} }));
const spans = memoryExporter.getFinishedSpans();
assert.strictEqual(spans.length, 1);
...
```

Reply:
What do you mean by never called?
This code part is the second argument to `assert.throws` which is executed upon the first argument throwing an exception.
It accepts an `err` parameter which is the exception thrown, and is used for the assertion on the span.

Reply:
Verified it again. If I change one of the assertions (for example, from 'localhost' to 'foo bar') and run `yarn test`, the test fails with the correct error:

```
  1) HttpInstrumentation
       enable()
         with good instrumentation options
           should have 1 ended span when request throw on bad "options" object:

      AssertionError [ERR_ASSERTION]: must be consistent (PEER_NAME and hostname)
      + expected - actual

      -localhost
      +foo bar
      
      at Object.assertSpan (test/utils/assertSpan.ts:109:12)
      at Object.<anonymous> (test/functionals/http-enable.test.ts:500:11)
      at expectedException (assert.js:633:26)
      at expectsError (assert.js:748:3)
      at Function.throws (assert.js:797:3)
      at Context.<anonymous> (test/functionals/http-enable.test.ts:483:16)
      at processImmediate (internal/timers.js:456:21)
      at process.callbackTrampoline (internal/async_hooks.js:120:14)
```

Reply:
Ahh yes, sorry.
Miscounted the parenthesis and besides that I was not aware that `assert.throws` can be used that way.